.circleci/config.yml

```diff
@@ -44,50 +25,38 @@ jobs:
       run-lint:
         type: boolean
         default: false
-      old-version:
-        type: boolean
-        default: false
-      work-dir:
-        type: string
-        default: ~/project
   
     docker:
       - image: <<parameters.docker-image>>
         environment:
           CIRCLE_TEST_REPORTS: /tmp/circle-reports
 
-    working_directory: <<parameters.work-dir>>
-
     steps:
       - checkout
 
       - run:
           name: install go-junit-report
           command: go get -u github.com/jstemmer/go-junit-report
 
-      - unless:
-          condition: <<parameters.old-version>>
-          steps:
-            - run:
-                name: build and test
-                command: make test | tee output.txt
-            - run:
-                name: process test results
-                command: |
-                  mkdir -p $CIRCLE_TEST_REPORTS
-                  go-junit-report < output.txt > $CIRCLE_TEST_REPORTS/junit.xml
-                when: always
-            - store_test_results:
-                path: /tmp/circle-reports
-      
-      - when:
-          condition: <<parameters.old-version>>
-          steps:
-            - run:
-                name: build only (no test in old Go versions)
-                command: make
+      - run:
+          name: build and test
+          command: make test | tee output.txt
+      - run:
+          name: process test results
+          command: |
+            mkdir -p $CIRCLE_TEST_REPORTS
+            go-junit-report < output.txt > $CIRCLE_TEST_REPORTS/junit.xml
+          when: always
+      - store_test_results:
+          path: /tmp/circle-reports
```

Comment:
This block isn't changed, it's just that I removed the obsolete "old-version" conditional.